Kitodo/src/main/java/org/kitodo/production/migration/NewspaperProcessesMigrator.java

```diff
@@ -0,0 +1,745 @@
+     */
+    private Iterator<Entry<String, IncludedStructuralElement>> yearsIterator;
+
+    /**
+     * The next year process to create.
+     */
+    private Entry<String, IncludedStructuralElement> yearToCreate;
+
+    /**
+     * Creates a new process migrator.
+     *
+     * @param batchTransfer
+     *            transfers Production v. 2 newspaper batch
+     */
+    public NewspaperProcessesMigrator(BatchDTO batchTransfer) {
+        this.batchNumber = batchTransfer.getId();
+        this.processesTransfer = batchTransfer.getProcesses();
+    }
+
+    /**
+     * Returns all newspaper batches.
+     *
+     * @return all newspaper batches
+     * @throws DataException
+     *             if a search engine error occurs
+     * @throws DAOException
+     *             if a process cannot be load from the database
+     * @throws IOException
+     *             if an I/O error occurs when accessing the file system
+     */
+    public static List<BatchDTO> getNewspaperBatches() throws DataException, DAOException, IOException {
+        final BatchService batchService = ServiceManager.getBatchService();
+        final ProcessService processService = ServiceManager.getProcessService();
+        final FileService fileService = ServiceManager.getFileService();
+
+        List<BatchDTO> newspaperBatches = new ArrayList<>();
+        for (BatchDTO batchTransfer : batchService.findAll()) {
+            if (isNewspaperBatch(batchTransfer, processService, fileService)) {
+                newspaperBatches.add(batchTransfer);
+            }
+        }
+        return newspaperBatches;
+    }
+
+    /**
+     * Returns whether the batch in the transfer object is a newspaper batch. A
+     * batch is a newspaper batch, if all of its processes are newspaper
+     * processes. A process is a newspaper process if it has a
+     * {@code meta_year.xml} file.
+     *
+     * @param batchTransfer
+     *            object transferring a batch
+     * @param processService
+     *            service to access processes
+     * @param fileService
+     *            service to access files
+     * @return whether the batch is a newspaper batch
+     * @throws DAOException
+     *             if a process cannot be load from the database
+     * @throws IOException
+     *             if an I/O error occurs when accessing the file system
+     */
+    private static boolean isNewspaperBatch(BatchDTO batchTransfer, ProcessService processService,
+            FileService fileService) throws DAOException, IOException {
+
+        logger.trace("Examining batch {}...", batchTransfer.getTitle());
+        boolean newspaperBatch = true;
+        for (ProcessDTO processTransfer : batchTransfer.getProcesses()) {
+            Process process = processService.getById(processTransfer.getId());
+            URI yearMetadata = insertBeforeXml(fileService.getMetadataFilePath(process), APPENDIX_YEAR);
+            if (!fileService.fileExist(yearMetadata)) {
```

Comment:
there should be a method in Process Service like "processOwnsYearXml" or something
Please try to avoid "inserting something before xml"